functions/calculate-score/index.js

```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const { performance } = require('perf_hooks')
+
+const v1 = require('./v1')
+
+module.exports = async function (context) {
+  const start = performance.now()
+  try {
+    await v1.process(context)
+  } catch (error) {
+    context.log.error(`calculate-score: ERROR: ${error.message}`)
+    throw error
```

Comment:
do we really want to throw this if there is simply no check window to calculate?

Reply:
@GuyHarwood within `v1` I am not throwing any error at all in case a check window is not detected but instead I simply return.
This is a standard try/catch we use in all the functions in case any data method throws an exception.
I would consider it quite important to keep this structure on the index level.

Reply:
the concern i have is that we are assuming that this specific situation will never happen because it's coded perfectly to only all the stored procedure when there is a check window active.  I am not suggesting you swallow every error, but to handle the 'no check window' errors more gracefully.

Reply:
you can use raiserror to return a specific custom error code and handle it based upon that

Reply:
@GuyHarwood in case a check window is not detected at the beginning of `v1` I am displaying this log message:
`calculate-score: Live check window not found`
There is no error thrown at all as I simply log and return.
Would you anticipate a different approach?

Reply:
I believe the store procedure call or any other data layer service call should only throw in case of a connection issue to the database.
This is a case why the catch exists on the top level of the function.

Reply:
i am clear about what is going on and why.  but you just have a generic catch when you could be doing something useful and informative in the 'exceptional' circumstance when you *do* execute the function outside a check window, for whatever reason

Reply:
the irony is that you are handling the 'exception' but not thinking that somehow there could be an exceptional situation where you execute it and it raises an error that you don't need to throw

Reply:
@GuyHarwood changed severity on the SP as discussed